app/init/draft_upload_manager.test.ts

```diff
@@ -0,0 +1,509 @@
+            // eslint-disable-next-line no-await-in-loop
+            await addFilesToDraft(validServerUrl, channelIds[i], rootIds[i], [file]);
+            manager.prepareUpload(validServerUrl, file, channelIds[i], rootIds[i], 0);
+        }
+
+        (mockClient.apiClient.upload as jest.Mock).mockRestore();
+
+        for (let i = 0; i < 3; i++) {
+            progressFunc[fileUrls[i]](0.1, bytesReads[i]);
+        }
+
+        AppState.currentState = 'background';
+        eventListener!('background');
+
+        // Wait for other promises to finish
+        await new Promise(process.nextTick);
+
+        for (let i = 0; i < 3; i++) {
+            // eslint-disable-next-line no-await-in-loop
+            const draft = await queryDraft(operator.database, channelIds[i], rootIds[i]);
+            expect(draft?.files.length).toBe(1);
+            expect(draft?.files[0].bytesRead).toBe(bytesReads[i]);
+        }
+
+        // Add progress inside the time window where it should not store in background
+        spyNow.mockImplementation(() => timeNotStore);
+
+        for (let i = 0; i < 3; i++) {
+            progressFunc[fileUrls[i]](0.1, bytesReadsNotStore[i]);
+        }
+
+        // Wait for other promises to finish
+        await new Promise(process.nextTick);
+
+        for (let i = 0; i < 3; i++) {
+            // eslint-disable-next-line no-await-in-loop
+            const draft = await queryDraft(operator.database, channelIds[i], rootIds[i]);
+            expect(draft?.files.length).toBe(1);
+            expect(draft?.files[0].bytesRead).toBe(bytesReads[i]);
+        }
+
+        // Add progress inside the time window where it should store in background
+        spyNow.mockImplementation(() => timeStore);
+
+        for (let i = 0; i < 3; i++) {
+            progressFunc[fileUrls[i]](0.1, bytesReadsStore[i]);
+
+            // Wait for other promises to finish (if not, watermelondb complains)
+            // eslint-disable-next-line no-await-in-loop
+            await new Promise(process.nextTick);
+        }
+
+        for (let i = 0; i < 3; i++) {
+            // eslint-disable-next-line no-await-in-loop
+            const draft = await queryDraft(operator.database, channelIds[i], rootIds[i]);
+            expect(draft?.files.length).toBe(1);
+            expect(draft?.files[0].bytesRead).toBe(bytesReadsStore[i]);
+        }
+
+        for (let i = 0; i < 3; i++) {
+            progressFunc[fileUrls[i]](0.1, bytesReadBeforeActive[i]);
+        }
+
+        AppState.currentState = 'active';
+        eventListener!('active');
+
+        for (let i = 0; i < 3; i++) {
+            // eslint-disable-next-line no-await-in-loop
+            const draft = await queryDraft(operator.database, channelIds[i], rootIds[i]);
+            expect(draft?.files.length).toBe(1);
+            expect(draft?.files[0].bytesRead).toBe(bytesReadsStore[i]);
+        }
+
+        spyNow.mockClear();
```

Comment:
``` Why do we clear it at the end ?
        spyNow.mockClear();
```

Reply:
It is cleanup. We don't do it on the other functions because we are mocking the implementation just once. Here instead we are mocking the implementation forever, and some cleanup should be in place.